### PR TITLE
Page visibility change cascades to all children

### PR DESCRIPTION
**Pull request by andrearutrecht**
Changing the visibility of a page now updates the visibility of all child pages as well.

---

## PR Checklist

-   [x] Verify if the change requires a changeset. See [CONTRIBUTING.md](https://github.com/vivid-planet/comet/blob/HEAD/CONTRIBUTING.md)
-   [x] Link to the respective task if one exists: PHSB2C-3155
-   [x] Provide screenshots/screencasts if the change contains visual changes

<details>
    <summary>Screenshots/screencasts</summary>

https://github.com/user-attachments/assets/7b525051-b8e0-44c2-8cef-ded1dd893436

</details>

### CODE REVIEW

**Comment by nsams** [comment]
I don't think this is a good idea.

- the user might not want to publish all child pages when publishing the parent page
- the user might want to keep the published/unpublished state when unpublishing the parent page
- it is Ok for archiving though

The underlaying issue (as described in PHSB2C-3155) is something different: Search indexes published pages even if a parent page is unpublished. When querying for pages to index you have to respect parent states.

Note: this PR still allows the situation that causes the issue from PHSB2C-3155 (although not that often anymore)

File: packages/admin/cms-admin/src/pages/pageTree/PageVisibility.tsx [hunk 1]
```diff
@@ -59,7 +60,7 @@ const PageVisibility = ({ page }: PageVisibilityProps): React.ReactElement => {
                     slug: page.slug,
                 },
             },
-        });
+        }).then(() => client.refetchQueries({ include: ["Pages"] }));
```
Comment (thomasdax98) [inline packages/admin/cms-admin/src/pages/pageTree/PageVisibility.tsx:63]: You can do that simpler:

```suggestion
            refetchQueries: ["Pages"],
        });
```

**Review by thomasdax98 — COMMENTED** [review]
I'm not sure if this is the behavior we want from a UX perspective. Accidentally publishing a page causes all sub pages to also be published (and lose their previous status). This increases error potential by a lot in my opinion.

Also, iterating through all the child nodes might be a performance problem if there are a lot of sub pages.

I think we have to talk this over in the COMET JFX

**Comment by thomasdax98** [comment]
Also, as Niko said, this should be handled correctly while creating the search index: 

>  Search indexes published pages even if a parent page is unpublished. When querying for pages to index you have to respect parent states.

(It's already handled correctly for pages in the site. If you unpublish the parent page, the child page is also invisible, although it hasn't the status in the admin)

**Comment by johnnyomair** [comment]
> It's already handled correctly for pages in the site

Are you sure? If I understand the code correctly, the visibility filter isn't considered in the pageTreeNodeByPath-Method anymore: https://github.com/vivid-planet/comet/blob/main/packages/api/cms-api/src/page-tree/page-tree-read-api.ts#L314-L318

**Comment by andrearutrecht** [comment]
> I'm not sure if this is the behavior we want from a UX perspective. Accidentally publishing a page causes all sub pages to also be published (and lose their previous status). This increases error potential by a lot in my opinion.
> 
> Also, iterating through all the child nodes might be a performance problem if there are a lot of sub pages.
> 
> I think we have to talk this over in the COMET JFX

I will wait for an update before working on this again, then.

**Comment by johnnyomair** [comment]
@andrearutrecht we discussed this yesterday and decided that we don't want this change. However, your specific problem should be solvable in the project. We can discuss this in person next week.

**Pull request closed without merging**